tests/scripts/mbed-build.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+
+# mbed-build.sh
+#
+# This file is part of Mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2019, Arm Limited, All Rights Reserved
+#
+# Purpose
+#
+# To run test builds of the Mbed OS module for all specified targets.
+# Usage: mbed-build <folder to import to> <optinally string of space seperated targets>
+# If no targets given as input, a set of default targets are tested, defined in
+# TARGETS variable.
+# The script uses a couple of environment variables:
+#     MBED_APP - The Mbed application to import and build. Default is mbed-os-example-tls.
+#     REMOTE_URL - The URL for the remote fork. Default is git@github.com:ARMmbed.
+
+DEFAULT_MBED_APP='mbed-os-example-tls'
+DEFAULT_REMOTE_URL="git@github.com:ARMmbed"
+DEFAULT_TARGETS="K64F NUCLEO_F429ZI"
+print_usage()
+{
+    echo "\nUsage: mbed-build <folder to import to> <optinally string of space seperated targets>"
+    echo "If no targets given as input, The following default targets are built:"
+    echo "$DEFAULT_TARGETS"
+    echo "The script uses a couple of environment variables:"
+    echo "    MBED_APP - The Mbed application to import and build. Default is $DEFAULT_MBED_APP."
+    echo "    REMOTE_URL - The URL for the remote fork. Default is $DEFAULT_REMOTE_URL."
+}
+
+if [ -z "$1" ]
+then
+    echo "expected folder to import not given as parameter"
+    print_usage
+    exit 1
+fi
+OUT_OF_SOURCE_DIR=$1
+
+if [ -z "$MBED_APP" ]
+then
+    MBED_APP=$DEFAULT_MBED_APP
+fi
+if [ -z "$REMOTE_URL" ]
+then
+    REMOTE_URL=$DEFAULT_REMOTE_URL
+fi
+if [ $# -eq 1 ]
+then
+    TARGETS=$DEFAULT_TARGETS
```

Comment:
I think that it is better to use the `:=` operator and variable expansion instead of these `DEFAULT_*` variables. Take a look at [all.sh](https://github.com/ARMmbed/mbedtls/blob/development/tests/scripts/all.sh#L127) for an example or the [standard](http://pubs.opengroup.org/onlinepubs/007904975/utilities/xcu_chap02.html).

Reply:
Thanks for the pointer! I was looking how to do this, but all I could find was that this is a Make syntax

Reply:
However:
1. I think it is relevant only for the variables that are defined in the environment, and not given as input parameters.
2. I still think we need the `DEFAULT_*` variables, for the usage printing